### ISSUE TITLE
[BUG] Escape regex in Metrics and Display-Series overrides

### ISSUE THREAD

**Issue by tankywoo** [grafana/grafana#7703]
Please include this information:
- What Grafana version are you using?  **grafana-4.1.2**
- What datasource are you using?  **influxdb**
- What OS are you running grafana on?  **gentoo**

- What did you do?

I defined a template `$mountpoint` represent the mountpoint of disk, such as `/`, '/home', '/boot/efi'。

and the metrics \:

```
SELECT mean("inode")  FROM "disk" WHERE $timeFilter AND host = 'server1' AND mountpoint =~ /^$mountpoint$/ GROUP BY time($interval) fill(null)
```

and alias is `$mountpoint (inode)`。

First, above is not work, because the mountpoint with slash not escaped in template, I search issues and found #2373 said it has been fixed, but what I tested is not.

so, I tried to escape manually，convert `mountpoint =~ /^$mountpoint$/` to `mountpoint =~ /^\$mountpoint$/` with a **explicit backslash**, this works for `/` and `/home`, but can't work for `/boot/efi` because there is a slash in the middle 😢 。

Another bug is when I defined the alias as above, If I define **Display -> Series overrides**, the items in the selected box, for example `/home (inode)` not work, even if I also add a backslash to escape. I found if the mountpoint name (as slash) not in the first character, it will work, alias such as `inode $mountpoint`.

**Comment by torkelo** [comment]
is your query using the query editor or raw query (text editor mode)? 

What is the variable options? (All, Multi-value etc)

**Comment by torkelo** [comment]
in raw query mode only variables query Variables with `All` or `Multi-value` enabled are regex escaped

**Comment by tankywoo** [comment]
@torkelo 

for first problem, regex escape why not support raw query without option enabled?

for ensond problem, please confirm ?

**Comment by thuck** [comment]
The pull #7734 should fix the problem with the / slash.

**Comment by tankywoo** [comment]
@thuck Had this been released? I want to upgrade and check.

**Comment by torkelo** [comment]
@tankywoo because regex escape is mostly only required when having multiple values. Since there is no option to turn it on/off Grafana tries to turn on regex escape only when it is required.

**Comment by daniellee** [comment]
@tankywoo  #7734 is not released yet. It will be included in the [4.2.0 release](https://github.com/grafana/grafana/milestone/47)

**Comment by rotti** [comment]
I have a similiar usecase and would like to have regex escaping without using ALL/Multi-Value. I want to to show several partitions/mountpoints with their used and available sizes. Because some partitions do have a slash inside (like '/var/log') I need the escaping.

But give the user the options to multi-select partions seems not to be the right thing for me, when it comes to different partions.

Grafana: Version 4.5.2 (commit: ec2b0fe6b)
InfluxDB: InfluxDB v0.13.0
